nsot/util/core.py

```diff
@@ -116,6 +116,35 @@ def slugify(s):
     return s
 
 
+def slugify_interface(interface=None, device_hostname=None, name=None,
```

Comment:
I'm picking up what you're putting down, but I think for that to work it would have to be a class method.

I'll experiment a little.